### PR TITLE
Fix CI bug

### PR DESCRIPTION
**Pull request by gdynusa**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        cp /tmp/docker/* /usr/bin
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        chmod +w /usr/bin
        cp /tmp/docker/* /usr/bin
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -19,7 +19,8 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        chmod +w /usr/bin
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -19,7 +19,8 @@ references:
...
+        cp /tmp/docker/* /usr/bin
```
Comment (gdynusa) [inline .circleci/config.yml:23]:
```suggestion
        sudo cp /tmp/docker/* /usr/bin
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        sudo cp /tmp/docker/* /usr/bin
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        sudo mv /tmp/docker/* /usr/bin
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -19,7 +19,6 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
         pacman --noconfirm --needed -Syu minio
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman -Sy archlinux-keyring
        pacman --noconfirm --needed -Syu minio
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman -Sy archlinux-keyring
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman --noconfirm --needed -Sy archlinux-keyring
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman --noconfirm --needed -Sy archlinux-keyring
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver hkp://pool.sks-keyservers.net
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver hkp://pool.sks-keyservers.net
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver hkps://hkps.pool.sks-keyservers.net:443
```

File: .circleci/config.yml [hunk 10]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver hkps://hkps.pool.sks-keyservers.net:443
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver hkp://pgp.mit.edu:11371
```

File: .circleci/config.yml [hunk 11]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver hkp://pgp.mit.edu:11371
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver pgp.mit.edu
```

File: .circleci/config.yml [hunk 12]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver pgp.mit.edu
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver hkp://ipv4.pool.sks-keyservers.net
```

File: .circleci/config.yml [hunk 13]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver hkp://ipv4.pool.sks-keyservers.net
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver hkp://keyserver.pgp.com
```

Comment (gdynusa) [inline .circleci/config.yml]:
```suggestion
        pacman-key --refresh-keys --keyserver keys.openpgp.org
```

File: .circleci/config.yml [hunk 14]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver hkp://keyserver.pgp.com
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver keys.openpgp.org
```

Comment (gdynusa) [inline .circleci/config.yml]:
```suggestion
        pacman-key --refresh-keys --keyserver keyserver.ubuntu.com
```

File: .circleci/config.yml [hunk 15]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver keys.openpgp.org
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
        pacman-key --refresh-keys --keyserver keyserver.ubuntu.com
```

File: .circleci/config.yml [hunk 16]
```diff
@@ -19,7 +19,7 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
+        pacman-key --refresh-keys --keyserver keyserver.ubuntu.com
```
Comment (gdynusa) [inline .circleci/config.yml:22]:
```suggestion
```

**Comment by gdynusa** [comment]
@petrbel we need new emloop docker image as the newest is 3 yrs old (https://hub.docker.com/r/iterait/emloop/tags) and pacman -Syu will always fail. then we can move to another problems.

File: .circleci/config.yml [hunk 17]
```diff
@@ -19,8 +19,8 @@ references:
         VER="17.11.0-ce"
         curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/test/x86_64/docker-$VER.tgz
         tar -xz -C /tmp -f /tmp/docker-$VER.tgz
-        mv /tmp/docker/* /usr/bin
-        pacman --noconfirm --needed -Syu minio
+        pacman --noconfirm --needed -Syu minio python-babel python-click python-matplotlib python-numpy python-requests python-ruamel-yaml python-testfixtures python-tabulate
+        pip install --no-cache-dir git+https://github.com/iterait/emloop.git
```
Comment (gdynusa) [inline .circleci/config.yml:23]:
```suggestion
        pip3 install --no-cache-dir git+https://github.com/iterait/emloop.git
```

File: .circleci/config.yml [hunk 18]
```diff
@@ -19,8 +19,8 @@ references:
...
         mkdir examples/docker/emloop_example/ssh
         pip install .
```
Comment (gdynusa) [inline .circleci/config.yml:25]:
```suggestion
        pip3 install .
```

File: .circleci/config.yml [hunk 19]
```diff
@@ -19,8 +19,8 @@ references:
...
         pip install '.[tests]'
```
Comment (gdynusa) [inline .circleci/config.yml:26]:
```suggestion
        pip3 install '.[tests]'
```

File: setup.py [hunk 20]
```diff
@@ -34,17 +34,17 @@
...
+        'ruamel.yaml==0.17.17',
+        'requests==2.26.0',
+        'schematics==2.1.1',
+        'aiohttp==3.7.4',
+        'aiohttp-cors==0.7.0',
         'emloop>=0.2',
-        'apistrap>=0.9.1',
-        'minio>=4.0',
+        'apistrap==0.9.11',
+        'minio==5.0.6',
```
Comment (Teyras) [inline setup.py:47]: Is it really necessary to be this specific? Not that it matters much...

**Review by Teyras — APPROVED** [review]

File: setup.py (hunk 20 above)
Comment (gdynusa) [reply setup.py:47]: @teyras well, kinda yes. It downloaded minio 7 something and didn't work. So i took packages that I already proven in isletnet as working together and specified them hereb as well.

**Pull request merged**